### ISSUE TITLE
Problem with sklearn GridSearchCV

### ISSUE THREAD

**Issue by vabatista** [dnouri/nolearn#216]
When I try to use a NeuralNet object with GridSearchCV, in console output is printed for each epoch, repeated (train, validation, ..) for each fold (cv).
It doesn't appear that cross validation is really happening.

**Comment by dnouri** [comment]
Can you give a more precise description of what you're seeing and what
you expect.  You're saying it *is* printing stuff out for each fold.
How does that mean CV is *not* happening?

**Comment by vabatista** [comment]
I thought CV (cross validation) wasn't working, but now I realize that the problem is only in output. Each epoch is been printed as many folds you define in gridsearch.

Here is my code:

```
net1 = NeuralNet(layers=layers,
     input_shape=(None, 62), ##62 is the resulting number of columns after feature union step in pipeline
     output_num_units=1,
     dense0_num_units=64,
     dropout0_p=0.3,
     dense1_num_units=64,
     dropout1_p=0.3,
     update_learning_rate=0.01,
     eval_size=0.01,
     verbose=1,
     output_nonlinearity=None,
     update=adagrad,
     max_epochs=5, regression=True
     #,custom_score=('rmse', lambda x, y: fmean_squared_error(x, y))
    ) 
clf = pipeline.Pipeline([('union', fu), ('nn', net1)])

param_grid={"nn__dense0_num_units": [64,128]}

nnModel = grid_search.GridSearchCV(estimator = clf, param_grid = param_grid, 
                                   n_jobs = 1, verbose=100 , cv = 3, scoring=RMSE)
nnModel.fit(X_train, Y2)
```

Here is the output:

Fitting 3 folds for each of 2 candidates, totalling 6 fits
[CV] nn__dense0_num_units=64 .........................................
  input             	(None, 62)          	produces      62 outputs
  dense0            	(None, 64)          	produces      64 outputs
  dropout0          	(None, 64)          	produces      64 outputs
  dense1            	(None, 64)          	produces      64 outputs
  dropout1          	(None, 64)          	produces      64 outputs
  output            	(None, 1)           	produces       1 outputs
  epoch      train loss    valid loss    train/val  dur
-------  --------------  ------------  -----------  -----
      1  10979528.56605   12162.80202    902.71375  0.46s
  epoch      train loss    valid loss    train/val  dur
-------  --------------  ------------  -----------  -----
      1  10979528.56605   12162.80202    902.71375  0.46s
  epoch      train loss    valid loss    train/val  dur
-------  --------------  ------------  -----------  -----
      1  10979528.56605   12162.80202    902.71375  0.46s
      2  671686.51262     692.35118    970.15291  0.51s
      2  671686.51262     692.35118    970.15291  0.51s
      2  671686.51262     692.35118    970.15291  0.51s
      3  333892.75287     163.44627   2042.82880  0.53s
      3  333892.75287     163.44627   2042.82880  0.53s
      3  333892.75287     163.44627   2042.82880  0.53s
      4  219504.21736     145.34174   1510.26281  0.53s
      4  219504.21736     145.34174   1510.26281  0.53s
      4  219504.21736     145.34174   1510.26281  0.53s
      5  154433.03904      63.34803   2437.85085  0.53s
      5  154433.03904      63.34803   2437.85085  0.53s
      5  154433.03904      63.34803   2437.85085  0.53s
[CV] ............... nn__dense0_num_units=64, score=-6.943987 -  17.8s
[Parallel(n_jobs=1)]: Done   1 tasks       | elapsed:   17.9s
[CV] nn__dense0_num_units=64 .........................................
  input             	(None, 62)          	produces      62 outputs
  dense0            	(None, 64)          	produces      64 outputs
  dropout0          	(None, 64)          	produces      64 outputs
  dense1            	(None, 64)          	produces      64 outputs
  dropout1          	(None, 64)          	produces      64 outputs
  output            	(None, 1)           	produces       1 outputs
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6262720.99690    1523.99665   4109.40602  0.47s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6262720.99690    1523.99665   4109.40602  0.47s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6262720.99690    1523.99665   4109.40602  0.47s
      2  367048.38152      42.32774   8671.58032  0.49s
      2  367048.38152      42.32774   8671.58032  0.49s
      2  367048.38152      42.32774   8671.58032  0.49s
      3  189420.36715      34.15182   5546.42011  0.49s
      3  189420.36715      34.15182   5546.42011  0.49s
      3  189420.36715      34.15182   5546.42011  0.49s
      4  128853.32059     127.99189   1006.73036  0.48s
      4  128853.32059     127.99189   1006.73036  0.48s
      4  128853.32059     127.99189   1006.73036  0.48s
      5   96134.20645      82.43097   1166.23890  0.49s
      5   96134.20645      82.43097   1166.23890  0.49s
      5   96134.20645      82.43097   1166.23890  0.49s
[CV] .............. nn__dense0_num_units=64, score=-13.366080 -  18.3s
[Parallel(n_jobs=1)]: Done   2 tasks       | elapsed:   36.3s
[CV] nn__dense0_num_units=64 .........................................
  input             	(None, 62)          	produces      62 outputs
  dense0            	(None, 64)          	produces      64 outputs
  dropout0          	(None, 64)          	produces      64 outputs
  dense1            	(None, 64)          	produces      64 outputs
  dropout1          	(None, 64)          	produces      64 outputs
  output            	(None, 1)           	produces       1 outputs
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  9920004.16525    2360.27555   4202.90086  0.46s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  9920004.16525    2360.27555   4202.90086  0.46s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  9920004.16525    2360.27555   4202.90086  0.46s
      2  572055.89747     284.75789   2008.92026  0.49s
      2  572055.89747     284.75789   2008.92026  0.49s
      2  572055.89747     284.75789   2008.92026  0.49s
      3  293684.28688      30.99259   9475.95221  0.52s
      3  293684.28688      30.99259   9475.95221  0.52s
      3  293684.28688      30.99259   9475.95221  0.52s
      4  188502.34889       5.78858  32564.53344  0.51s
      4  188502.34889       5.78858  32564.53344  0.51s
      4  188502.34889       5.78858  32564.53344  0.51s
      5  137167.67857       1.04053  131824.30963  0.51s
      5  137167.67857       1.04053  131824.30963  0.51s
      5  137167.67857       1.04053  131824.30963  0.51s
[CV] ............... nn__dense0_num_units=64, score=-3.492526 -  17.6s
[Parallel(n_jobs=1)]: Done   3 tasks       | elapsed:   54.1s
[CV] nn__dense0_num_units=128 ........................................
  input             	(None, 62)          	produces      62 outputs
  dense0            	(None, 128)         	produces     128 outputs
  dropout0          	(None, 128)         	produces     128 outputs
  dense1            	(None, 64)          	produces      64 outputs
  dropout1          	(None, 64)          	produces      64 outputs
  output            	(None, 1)           	produces       1 outputs
  epoch      train loss    valid loss    train/val  dur
-------  --------------  ------------  -----------  -----
      1  17537990.49285   13487.39942   1300.32410  0.69s
  epoch      train loss    valid loss    train/val  dur
-------  --------------  ------------  -----------  -----
      1  17537990.49285   13487.39942   1300.32410  0.69s
  epoch      train loss    valid loss    train/val  dur
-------  --------------  ------------  -----------  -----
      1  17537990.49285   13487.39942   1300.32410  0.69s
      2  808360.26660     400.34631   2019.15256  0.71s
      2  808360.26660     400.34631   2019.15256  0.71s
      2  808360.26660     400.34631   2019.15256  0.71s
      3  377622.55266     278.54546   1355.69450  0.71s
      3  377622.55266     278.54546   1355.69450  0.71s
      3  377622.55266     278.54546   1355.69450  0.71s
      4  218420.00615      71.32234   3062.43459  0.73s
      4  218420.00615      71.32234   3062.43459  0.73s
      4  218420.00615      71.32234   3062.43459  0.73s
      5  162244.77859      29.17208   5561.64546  0.74s
      5  162244.77859      29.17208   5561.64546  0.74s
      5  162244.77859      29.17208   5561.64546  0.74s
[CV] .............. nn__dense0_num_units=128, score=-4.654700 -  19.0s
[Parallel(n_jobs=1)]: Done   4 tasks       | elapsed:  1.2min
[CV] nn__dense0_num_units=128 ........................................
  input             	(None, 62)          	produces      62 outputs
  dense0            	(None, 128)         	produces     128 outputs
  dropout0          	(None, 128)         	produces     128 outputs
  dense1            	(None, 64)          	produces      64 outputs
  dropout1          	(None, 64)          	produces      64 outputs
  output            	(None, 1)           	produces       1 outputs
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6636860.93942    8868.99595    748.32157  0.71s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6636860.93942    8868.99595    748.32157  0.71s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6636860.93942    8868.99595    748.32157  0.71s
      2  263834.19599    1176.64924    224.22502  0.71s
      2  263834.19599    1176.64924    224.22502  0.71s
      2  263834.19599    1176.64924    224.22502  0.71s
      3  124377.47467     258.64925    480.87313  0.70s
      3  124377.47467     258.64925    480.87313  0.70s
      3  124377.47467     258.64925    480.87313  0.70s
      4   81645.06365      55.63556   1467.49770  0.73s
      4   81645.06365      55.63556   1467.49770  0.73s
      4   81645.06365      55.63556   1467.49770  0.73s
      5   57833.17669       4.37613  13215.60848  0.73s
      5   57833.17669       4.37613  13215.60848  0.73s
      5   57833.17669       4.37613  13215.60848  0.73s
[CV] .............. nn__dense0_num_units=128, score=-2.024193 -  18.9s
[Parallel(n_jobs=1)]: Done   5 tasks       | elapsed:  1.5min
[CV] nn__dense0_num_units=128 ........................................
  input             	(None, 62)          	produces      62 outputs
  dense0            	(None, 128)         	produces     128 outputs
  dropout0          	(None, 128)         	produces     128 outputs
  dense1            	(None, 64)          	produces      64 outputs
  dropout1          	(None, 64)          	produces      64 outputs
  output            	(None, 1)           	produces       1 outputs
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  9974989.52735     943.00505  10577.87493  0.65s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  9974989.52735     943.00505  10577.87493  0.65s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  9974989.52735     943.00505  10577.87493  0.65s
      2  494142.18502       6.00192  82330.69224  0.70s
      2  494142.18502       6.00192  82330.69224  0.70s
      2  494142.18502       6.00192  82330.69224  0.70s
      3  223168.78098       5.99686  37214.29700  0.70s
      3  223168.78098       5.99686  37214.29700  0.70s
      3  223168.78098       5.99686  37214.29700  0.70s
      4  138729.18603       5.99289  23148.97534  0.71s
      4  138729.18603       5.99289  23148.97534  0.71s
      4  138729.18603       5.99289  23148.97534  0.71s
      5   95238.91932       5.98838  15903.96329  0.71s
      5   95238.91932       5.98838  15903.96329  0.71s
      5   95238.91932       5.98838  15903.96329  0.71s
[CV] .............. nn__dense0_num_units=128, score=-2.322129 -  18.6s
[Parallel(n_jobs=1)]: Done   6 tasks       | elapsed:  1.9min
[Parallel(n_jobs=1)]: Done   6 out of   6 | elapsed:  1.9min finished
  input             	(None, 62)          	produces      62 outputs
  dense0            	(None, 128)         	produces     128 outputs
  dropout0          	(None, 128)         	produces     128 outputs
  dense1            	(None, 64)          	produces      64 outputs
  dropout1          	(None, 64)          	produces      64 outputs
  output            	(None, 1)           	produces       1 outputs
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6227028.51860     349.46734  17818.62795  1.01s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6227028.51860     349.46734  17818.62795  1.01s
  epoch     train loss    valid loss    train/val  dur
-------  -------------  ------------  -----------  -----
      1  6227028.51860     349.46734  17818.62795  1.01s
      2  193864.41800      33.23201   5833.66517  1.08s
      2  193864.41800      33.23201   5833.66517  1.08s
      2  193864.41800      33.23201   5833.66517  1.08s
      3  102871.49170      11.57513   8887.28601  1.03s
      3  102871.49170      11.57513   8887.28601  1.03s
      3  102871.49170      11.57513   8887.28601  1.03s
      4   68453.11836       6.93505   9870.60766  1.00s
      4   68453.11836       6.93505   9870.60766  1.00s
      4   68453.11836       6.93505   9870.60766  1.00s
      5   53885.64029       3.70795  14532.44578  0.99s
      5   53885.64029       3.70795  14532.44578  0.99s
      5   53885.64029       3.70795  14532.44578  0.99s

**Comment by dnouri** [comment]
So are you saying only the output is broken when doing a `GridSearchCV`?  If that's the case, I think that's sort of unfortunate but expected.  Please reopen if there's another issue.
